File: ActorsTests/Tests/Demo/ActorSystemMock.m
//
//  ActorSystemMock.m
//  Actors
//
//  Created by Anastasiya Gorban on 8/13/15.
//  Copyright (c) 2015 Techery. All rights reserved.
//

#import "ActorSystemMock.h"
#import "Actors-Swift.h"

DTMainActorSystem * actorSystemMock() {
    DTMainActorSystem *actorSystem = [DTMainActorSystem mock];

    [actorSystem stub:@selector(configs) andReturn:[KWMock mockForProtocol:@protocol(Configs)]];
    [actorSystem stub:@selector(serviceLocator) andReturn:[[ServiceLocator alloc] initWithBuilder:^(ServiceLocator *locator) {

    }]];

    return actorSystem;
}

File: ActorsTests/Tests/Demo/ActorSystemMock.swift
//
//  ActorSystemMock.swift
//  Actors
//
//  Created by Anastasiya Gorban on 8/17/15.
//  Copyright (c) 2015 Techery. All rights reserved.
//

import Actors

class ActorSystemMock: DTMainActorSystem {
    override init() {
        let serviceLocator = ServiceLocator(builder: {_ in})
        super.init(configs: ConfigsMock(), serviceLocator: serviceLocator, builderBlock: {_ in})
    }
}

class ConfigsMock: Configs {
    @objc subscript(key: String) -> AnyObject? {
        get {return nil}
        set {}
    }
}

